File: thermo/EBRAgent/src/main/java/uk/ac/cam/cares/query/QueryTemplate.java
package uk.ac.cam.cares.query;
/**
 * 
 * @author NK510 (caresssd@hermes.cam.ac.uk)
 * 
 *
 */
public class QueryTemplate {

	/**
	 * 
	 * @param webLinIri
	 * @return the species IRIs that have given web link .
	 */
	public static String getSpeciesIriWtihGivenWebLink(String webLinkIri) {
		
		String query = 
				"SELECT distinct ?s ?speciesIri "
				+ "WHERE { "  
				+ "?s <http://www.theworldavatar.com/ontology/ontocompchem/ontocompchem.owl#hasUniqueSpeciesIRI> ?speciesIri . "
				+ "?speciesIri <"+ webLinkIri+"> . "  
				+ "}";
		
		return query ;
	}
	
	/**
	 * 
	 * @author NK510 (caresssd@hermes.cam.ac.uk)
	 * @return the species iri, species registry id, atomic bond, and geometry
	 * 
	 */
	public static String getSpeciesRegistryIDAtomicBondAndGeometry() {
		
		String query ="PREFIX OntoSpecies: <http://www.theworldavatar.com/ontology/ontospecies/OntoSpecies.owl#>"
				+ "SELECT ?species ?crid ?atomicBond ?geometry "
				+ "WHERE { "
				+ "?species OntoSpecies:casRegistryID ?crid . "
				+ "?species OntoSpecies:hasAtomicBond ?atomicBond . "
				+ "?species OntoSpecies:hasGeometry ?geometry . "
				+ "} LIMIT 1";
		
		return query;
	}
	
}

File: MOPTools/CBU_to_os_JSON/kgoperations/querytemplate.py
def spin_inchi_query(inchi_string, spin_number):
    query = """
    PREFIX OntoSpecies: <http://www.theworldavatar.com/ontology/ontospecies/OntoSpecies.owl#>
    PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
    SELECT ?speciesIRI ?Inchi ?Spin
    WHERE
    {
    ?speciesIRI rdf:type OntoSpecies:Species .
    ?speciesIRI OntoSpecies:inChI ?Inchi .
    FILTER((?Inchi) = "#INCHI")
    ?speciesIRI OntoSpecies:spinMultiplicity ?Spin .
	FILTER((?Spin) = "#SPIN")
    }

    """
    query = query.replace('#INCHI', inchi_string)
    query = query.replace('#SPIN', spin_number)
    return query